### PR TITLE
xcode-backend: fix include paths

### PR DESCRIPTION
**Pull request by jmoguillansky-gpsw**
Add include paths from dependencies

### CODE REVIEW

File: mesonbuild/backend/xcodebackend.py [hunk 1]
```diff
@@ -751,6 +753,8 @@ def generate_xc_build_configuration(self):
                     args = pargs + gargs + targs
                     if args:
                         langargs[langnamemap[lang]] = args
+                cargs = list(dict.fromkeys(cargs)) # Remove duplicates, and preserve order
```
Comment (dcbaker) [inline mesonbuild/backend/xcodebackend.py:756]: `dict.fromkeys()` isn't guaranteed to preserve order in python3.6, please use `mesonlib.OrderedSet` for this, it's also what we do elsewhere.

Also, is this actually safe? Sometimes arguments can be given multiple times, but actually matter, like inside a `--start-group/--end-group`.

File: mesonbuild/backend/xcodebackend.py [hunk 2]
```diff
@@ -722,6 +722,7 @@ def generate_xc_build_configuration(self):
                 if links_dylib:
                     dep_libs = ['-Wl,-search_paths_first', '-Wl,-headerpad_max_install_names'] + dep_libs
                 dylib_version = None
+                cargs = []
```
Comment (dcbaker) [inline mesonbuild/backend/xcodebackend.py:725]: I know it's a nit, but can you put this closer to where it's used?

**Review by dcbaker — CHANGES_REQUESTED** [review]

File: mesonbuild/backend/xcodebackend.py (hunk 1 above)
Comment (jmoguillansky-gpsw) [reply mesonbuild/backend/xcodebackend.py]: thanks, you're right. Updated.  sorry for the delay

File: mesonbuild/backend/xcodebackend.py (hunk 2 above)
Comment (jmoguillansky-gpsw) [reply mesonbuild/backend/xcodebackend.py]: updated

File: mesonbuild/backend/xcodebackend.py [hunk 3]
```diff
@@ -751,6 +753,7 @@ def generate_xc_build_configuration(self):
                     args = pargs + gargs + targs
                     if args:
                         langargs[langnamemap[lang]] = args
+                langargs['C'] += cargs
```
Comment (dcbaker) [inline mesonbuild/backend/xcodebackend.py:756]: Should this really go into `langargs['C']` and not into `langargs[langnamemap[lang]]`?

Sorry I just noticed this.

**Comment by dcbaker** [comment]
Other than that this look good to me.

File: mesonbuild/backend/xcodebackend.py (hunk 3 above)
Comment (jmoguillansky-gpsw) [reply mesonbuild/backend/xcodebackend.py:756]: we probably want to set these flags for C and C++.  I think the main concern is not to set these flags for other languages like objective-c, etc.

Comment (jmoguillansky-gpsw) [reply mesonbuild/backend/xcodebackend.py:756]: https://developer.apple.com/library/archive/documentation/DeveloperTools/Reference/XcodeBuildSettingRef/1-Build_Setting_Reference/build_setting_ref.html

xcode spec has "OTHER_CFLAGS" and "OTHER_CPLUSPLUSFLAGS".

Default value of "OTHER_CPLUSPLUSFLAGS" is "OTHER_CFLAGS" so these flags should apply for C and C++

Comment (dcbaker) [reply mesonbuild/backend/xcodebackend.py:756]: okay, That's probably fine for now then. If we run into a case where we need to apply them to ObjC we can handle that later.

**Review by dcbaker — APPROVED** [review]

**Comment by dcbaker** [comment]
Thanks, merged.

**Pull request merged**